sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -341,8 +340,33 @@ void CAFMaker::beginJob() {
 
 //......................................................................
 void CAFMaker::beginRun(art::Run& run) {
-  // fDetID = geom->DetId();
-  fDet = (Det_t)1;//(Det_t)fDetID;
+  // Heuristic method to determine the detector ID
+  const geo::GeometryCore* geom = lar::providerFrom<geo::Geometry>();
+
+  std::string gdml = geom->GDMLFile();
+  gdml = basename(gdml.c_str()); // strip directory part
```

Comment:
I suppose you are using POSIX `basename()` because linking to `<filesystem>` library with Clang 7 and GCC 9 is cumbersome?

Reply:
Honestly, I wasn't really aware of it. See new PR